generators/client/templates/angular/src/main/webapp/app/layouts/main/main.component.ts.ejs

```diff
@@ -16,96 +16,58 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 -%>
-import { Component,<%_ if (authenticationType === 'oauth2') { _%> OnDestroy,<%_ } _%> OnInit } from '@angular/core';
-<%_ if (!enableTranslation) { _%>
+import { Component, OnInit } from '@angular/core';
 import { Title } from '@angular/platform-browser';
-<%_ } _%>
 import { Router, ActivatedRouteSnapshot, NavigationEnd, NavigationError } from '@angular/router';
-<%_ if (authenticationType === 'oauth2') { _%>
-import { Subject } from 'rxjs';
-import { takeUntil } from 'rxjs/operators';
-<%_ } _%>
-
 <%_ if (enableTranslation) { _%>
-import { JhiLanguageHelper } from 'app/core/language/language.helper';
-<%_ } _%>
-<%_ if (authenticationType === 'oauth2') { _%>
-import { Account } from 'app/core/user/account.model';
-import { AccountService } from 'app/core/auth/account.service';
-import { StateStorageService } from 'app/core/auth/state-storage.service';
+import { TranslateService } from '@ngx-translate/core';
 <%_ } _%>
 
 @Component({
     selector: '<%= jhiPrefixDashed %>-main',
     templateUrl: './main.component.html'
 })
-export class <%=jhiPrefixCapitalized%>MainComponent implements OnInit<%_ if (authenticationType === 'oauth2') { _%>, OnDestroy<%_ } _%> {
-    <%_ if (authenticationType === 'oauth2') { _%>
-    _cleanup: Subject<any> = new Subject<any>();
-    <%_ } _%>
-
+export class MainComponent implements OnInit {
     constructor(
-        <%_ if (authenticationType === 'oauth2') { _%>
-        private accountService: AccountService,
-        private stateStorageService: StateStorageService,
-        <%_ } _%>
         <%_ if (enableTranslation) { _%>
-        private jhiLanguageHelper: JhiLanguageHelper,
-        <%_ } else { _%>
-        private titleService: Title,
+        private translateService: TranslateService,
         <%_ } _%>
+        private titleService: Title,
         private router: Router
     ) {}
 
-    private getPageTitle(routeSnapshot: ActivatedRouteSnapshot) {
-        let title: string = (routeSnapshot.data && routeSnapshot.data['pageTitle']) ? routeSnapshot.data['pageTitle'] : '<%= angularAppName %>';
+    private getPageTitle(routeSnapshot: ActivatedRouteSnapshot): string {
+        let title: string = (routeSnapshot.data && routeSnapshot.data['pageTitle']) ? routeSnapshot.data['pageTitle'] : '';
         if (routeSnapshot.firstChild) {
             title = this.getPageTitle(routeSnapshot.firstChild) || title;
         }
         return title;
     }
 
-    ngOnInit() {
-        this.router.events.subscribe((event) => {
+    private updateTitle(): void {
```

Comment:
Can we move the private functions under public ones ?

Reply:
Done